src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -128,13 +128,16 @@ protected override void PerformWork()
                                             break;
                                     }
                                 }
-
-                                // Directly execute the OneTimeFixtureTearDown for tests that
-                                // were skipped, failed or set to inconclusive in one time setup
-                                // unless we are aborting.
-                                if (Context.ExecutionStatus != TestExecutionStatus.AbortRequested)
-                                    PerformOneTimeTearDown();
                             }
+
+                            // Directly execute the OneTimeFixtureTearDown for tests that
+                            // were skipped, failed or set to inconclusive in one time setup
+                            // unless we are aborting.
+                            // This is important to do even when there are no child tests,
```

Comment:
Alright! Sounds like we may have a solution then?
How would you word it? A "for example, theories" part could maybe still be useful.